Gems/ScriptCanvas/Code/Include/ScriptCanvas/Libraries/Spawning/SpawnTicketInstance.cpp

```diff
@@ -0,0 +1,91 @@
+            }
+        }
+
+        if (auto behaviorContext = azrtti_cast<AZ::BehaviorContext*>(context))
+        {
+            behaviorContext->Class<SpawnTicketInstance>("SpawnTicketInstance")
+                           ->Constructor()
+                           ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Common)
+                           ->Attribute(AZ::Script::Attributes::Category, "Spawning")
+                           ->Attribute(AZ::Script::Attributes::Module, "Spawning")
+                           ->Attribute(AZ::Script::Attributes::EnableAsScriptEventParamType, true);
+
+            AZ::BehaviorClass* behaviorClass =
+                AZ::BehaviorContextHelper::GetClass(behaviorContext, azrtti_typeid<BehaviorClassReflection<SpawnTicketInstance>>());
+            AZ::BehaviorContext::ClassBuilder<BehaviorClassReflection<SpawnTicketInstance>> classBuilder(behaviorContext, behaviorClass);
+            CreateTypeAsMapValueHelper<float, SpawnTicketInstance>::AddMethod(&classBuilder);
+            CreateTypeAsMapValueHelper<AZStd::string, SpawnTicketInstance>::AddMethod(&classBuilder);
+
+            behaviorContext
+                ->Class<BehaviorClassReflection<SpawnTicketInstance>>(
+                    AZStd::string::format("ReflectOnDemandTargets_%s", Data::Traits<SpawnTicketInstance>::GetName().data()).data())
+                ->Attribute(AZ::Script::Attributes::ExcludeFrom, AZ::Script::Attributes::ExcludeFlags::All)
+                ->Attribute(AZ::Script::Attributes::Ignore, true)
+                // required to support Array<SpawnTicketInstance> variable type in Script Canvas
+                ->Method(
+                    "ReflectVector",
+                    [](const AZStd::vector<SpawnTicketInstance>&)
+                    {
+                    })
+                // required to support Map<String, SpawnTicketInstance> variable type in Script Canvas
+                ->Method(
+                    "MapStringToSpawnTicketInstance",
+                    [](const AZStd::unordered_map<Data::StringType, SpawnTicketInstance>&)
+                    {
+                    })
+                // required to support Map<Number, SpawnTicketInstance> variable type in Script Canvas
+                ->Method(
+                    "MapNumberToSpawnTicketInstance",
+                    [](const AZStd::unordered_map<Data::NumberType, SpawnTicketInstance>&)
+                    {
+                    });
+        }
+
+        HashContainerReflector<SpawnTicketInstance>::Reflect(context);
+    }
+    
+    SpawnTicketInstance::SpawnTicketInstance(const SpawnTicketInstance& rhs)
+        : m_ticket(rhs.m_ticket)
+    {
+    }
+
+    SpawnTicketInstance& SpawnTicketInstance::operator=(const SpawnTicketInstance& rhs)
+    {
+        m_ticket = rhs.m_ticket;
+        return *this;
+    }
```

Comment:
This implementations are not needed and can be done in the header using `= default`.